File: services/frontend/src/types/seller.ts
export interface Seller {
  id: string
  email: string
  first_name: string
  last_name: string
  middle_name?: string | null
  company_name: string
  phone: string
  created_at?: string
  updated_at?: string
}

export interface AuthResponse {
  access_token: string
  refresh_token: string
  token_type: string
  user_id: string
}

export interface SellerFormValues {
  email: string
  first_name: string
  last_name: string
  middle_name: string
  company_name: string
  phone: string
  password: string
  passwordConfirm: string
}

export type SellerFormErrors = Partial<Record<keyof SellerFormValues, string>>

File: services/b2b/schemas/seller.py
from uuid import UUID

from pydantic import BaseModel, EmailStr


class SellerInfoResponse(BaseModel):
	id: UUID
	first_name: str
	last_name: str
	middle_name: str

	company_name: str


class SellerInfoPatch(BaseModel):
	email: EmailStr
	phone: str
	first_name: str
	last_name: str
	middle_name: str
	company_name: str
